### PR TITLE
Converters Unit Tests

### PR DESCRIPTION
**Pull request by zinkoviktor**

### CODE REVIEW

File: NetCoreRestApi/tests/UnitTests/ConvertersTests/CategoryModelConverterConvertFrom.cs [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+using System.Linq;
+
+namespace UnitTests.ConvertersTests
+{
+    [TestClass]    
+    public class CategoryModelConverterConvertFrom
+    {
+        CategoryModelConverter converter;
+        CategoryModel categoryModel;
+        List<CategoryModel> categoryModels;
```
Comment (mdemiy) [inline NetCoreRestApi/tests/UnitTests/ConvertersTests/CategoryModelConverterConvertFrom.cs]: private members

File: NetCoreRestApi/tests/UnitTests/ConvertersTests/CategoryModelConverterConvertFrom.cs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            converter = new CategoryModelConverter();
+
+            categoryModel = new CategoryModel()
```
Comment (mdemiy) [inline NetCoreRestApi/tests/UnitTests/ConvertersTests/CategoryModelConverterConvertFrom.cs]: use local variables

File: NetCoreRestApi/tests/UnitTests/ConvertersTests/CategoryModelConverterConvertFrom.cs [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+                     Name = "",
+                     Description = null,
+                }
+            };
+        }
+        
+        [TestMethod]         
+        public void CorrectClassTypeConvertedToCategoryEntity()
+        {
+            // Arrange
```
Comment (mdemiy) [inline NetCoreRestApi/tests/UnitTests/ConvertersTests/CategoryModelConverterConvertFrom.cs]: use local variable for model instance

File: NetCoreRestApi/tests/UnitTests/ConvertersTests/CategoryModelConverterConvertFrom.cs [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+                {
+                     Id = 0,
+                     Name = "",
+                     Description = null,
+                }
+            };
+        }
+        
+        [TestMethod]         
+        public void CorrectClassTypeConvertedToCategoryEntity()
```
Comment (mdemiy) [inline NetCoreRestApi/tests/UnitTests/ConvertersTests/CategoryModelConverterConvertFrom.cs]: Name best practices

File: NetCoreRestApi/tests/UnitTests/ConvertersTests/CategoryModelConverterConvertFrom.cs [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+            // Act
+            var actualCategoryEntity = converter.ConvertFrom(categoryModel);
+
+            // Assert            
+            Assert.AreEqual(expectedDescription, actualCategoryEntity.Description);
+        }
+
+        [TestMethod]
+        public void CorrectClassTypeConvertedToCategoryEntityCollection()
```
Comment (mdemiy) [inline NetCoreRestApi/tests/UnitTests/ConvertersTests/CategoryModelConverterConvertFrom.cs]: name best practices

File: NetCoreRestApi/tests/UnitTests/BaseComparer.cs [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    public class BaseComparer<T> : IComparer
+    {
+        private Func<T, T, bool> _compareFunc;
+
+        public BaseComparer(Func<T, T, bool> compareFunc)
+        {
+            _compareFunc = compareFunc;
+        }
+
+        public int Compare(object x, object y)
```
Comment (mdemiy) [inline NetCoreRestApi/tests/UnitTests/BaseComparer.cs]: Give full name for x, y

File: NetCoreRestApi/tests/UnitTests/DataLayer.EF/Converters/CategoryModelConverterTests/ConvertFrom.cs [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+                entity1.Id.Equals(entity2.Id) &&
+                entity1.Name.Equals(entity2.Name) &&
+                entity1.Description.Equals(entity2.Description);
+
+            _entityComparer = new BaseComparer<CategoryEntity>(predicate);
+        }
+
+        [TestMethod]
+        [Description("Convert To CategoryEntity collection all items are not Null")]
+        public void AllItemsAreNotNull()
```
Comment (mdemiy) [inline NetCoreRestApi/tests/UnitTests/DataLayer.EF/Converters/CategoryModelConverterTests/ConvertFrom.cs]: Use _ as space between words

File: NetCoreRestApi/tests/UnitTests/DataLayer.EF/Converters/CategoryModelConverterTests/ConvertFrom.cs [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+using DataLayer.Models;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace UnitTests.DataLayer.EF.Converters.CategoryModelConverterTests
+{
+    [TestClass]    
+    public class ConvertFrom
```
Comment (mdemiy) [inline NetCoreRestApi/tests/UnitTests/DataLayer.EF/Converters/CategoryModelConverterTests/ConvertFrom.cs]: Use name CategoryModelConverter_ConverFrom_Test

File: NetCoreRestApi/tests/UnitTests/DataLayer.EF/Converters/CategoryModelConverterTests/ConvertFrom.cs [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    {
+        private IConverter<CategoryEntity, CategoryModel> _converter;
+        private IComparer _entityComparer;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            _converter = new CategoryModelConverter();
+
+            static bool predicate(CategoryEntity entity1, CategoryEntity entity2) =>
```
Comment (mdemiy) [inline NetCoreRestApi/tests/UnitTests/DataLayer.EF/Converters/CategoryModelConverterTests/ConvertFrom.cs]: Refactor this into private filed

File: NetCoreRestApi/tests/UnitTests/DataLayer.EF/Converters/CategoryModelConverterTests/ConvertFrom.cs [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+            };
+            var categoryModel2 = new CategoryModel()
+            {
+                Id = 0,
+                Name = "",
+                Description = ""
+            };
+            var categoryModels = new List<CategoryModel>()
+            {
+                categoryModel, categoryModel2
```
Comment (mdemiy) [inline NetCoreRestApi/tests/UnitTests/DataLayer.EF/Converters/CategoryModelConverterTests/ConvertFrom.cs]: use only new CategoryModel instead variables

File: NetCoreRestApi/tests/UnitTests/DataLayer.EF/Converters/CategoryModelConverterTests/ConvertFrom.cs [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+                Name = "Name 1 !@~#$%^&*()_+=-\\||'\"?/.><,",
+                Description = "Name 1 !@~#$%^&*()_+=-\\||'\"?/.><,"
+            };
+            var categoryEntity2 = new CategoryEntity()
+            {
+                Id = 0,
+                Name = "",
+                Description = ""
+            };            
+            var expected = new List<CategoryEntity> { categoryEntity, categoryEntity2 };
```
Comment (mdemiy) [inline NetCoreRestApi/tests/UnitTests/DataLayer.EF/Converters/CategoryModelConverterTests/ConvertFrom.cs]: the same use only new CategoryEntity instead variables

File: NetCoreRestApi/tests/UnitTests/DataLayer.EF/Converters/CategoryModelConverterTests/ConvertTo.cs [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+using DataLayer.Models;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace UnitTests.DataLayer.EF.Converters.CategoryModelConverterTests
+{
+    [TestClass]    
+    public class ConvertTo
```
Comment (mdemiy) [inline NetCoreRestApi/tests/UnitTests/DataLayer.EF/Converters/CategoryModelConverterTests/ConvertTo.cs]: the same - rename class name

File: NetCoreRestApi/tests/UnitTests/DataLayer.EF/Converters/CategoryModelConverterTests/ConvertTo.cs [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+    {
+        private IConverter<CategoryEntity, CategoryModel> _converter;
+        private IComparer _entityComparer;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            _converter = new CategoryModelConverter();
+
+            static bool predicate(CategoryModel model1, CategoryModel model2) =>
```
Comment (mdemiy) [inline NetCoreRestApi/tests/UnitTests/DataLayer.EF/Converters/CategoryModelConverterTests/ConvertTo.cs]: refactor to private filed

**Review by mdemiy — CHANGES_REQUESTED** [review]
Fix comments

File: NetCoreRestApi/tests/UnitTests/Helpers/BaseComparer.cs [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+        public BaseComparer(Func<T, T, bool> compareFunc)
+        {
+            _compareFunc = compareFunc;
+        }
+
+        public int Compare(object x, object y)
+        {
+            if(x.GetType() != typeof(T))
+            {
+                return -1;
```
Comment (mdemiy) [inline NetCoreRestApi/tests/UnitTests/Helpers/BaseComparer.cs]: Magic number, create constants

**Review by mdemii — APPROVED** [review]

**Pull request merged**